frame/rewards/src/lib.rs

```diff
@@ -92,6 +109,8 @@ decl_module! {
 
 		fn on_finalize() {
 			if let Some(author) = <Self as Store>::Author::get() {
+				let treasury_id = pallet_treasury::Module::<T>::account_id();
```

Comment:
We should have a `type DonationDestination = Get<T::AccountId>`

And then in the runtime, we can set this account as the `treasury_id`, then this pallet need not take any dependencies on `pallet_treasury`.